packages/flutter_test/lib/src/matchers.dart

```diff
@@ -245,6 +245,18 @@ Matcher isSameColorAs(Color color, {double threshold = colorEpsilon}) {
   return _ColorMatcher(color, threshold);
 }
 
+/// Asserts that the object is a [TextScaler] that reflects the user's font
+/// scale preferences, typically from the platform's accessibility settings.
+///
+/// In widget tests, the value of the system font scale preference can be
+/// changed via [TestPlatformDispatcher.textScaleFactorTestValue].
+///
+/// If `withScaleFactor` is specified and non-null, this matcher also asserts
+/// that the [TextScaler]'s' `textScaleFactor` equals `withScaleFactor`.
+Matcher isSystemTextScaler({ double? withScaleFactor }) {
```

Comment:
What is a non-system text scaler? Can you add something to the docs that describes the difference between a system text scaler and a non-system one?

And for my own understanding: Why should I care in my tests whether the text scaler is system or not?